client/src/state/secure.rs

```diff
@@ -74,32 +79,32 @@ impl SecureClient {
     }
 
     /// Resolves a location to a `VaultId` and a `RecordId`
-    pub fn resolve_location<L: AsRef<Location>>(&self, l: L) -> (VaultId, RecordId) {
+    pub fn resolve_location<L: AsRef<Location>>(l: L) -> (VaultId, RecordId) {
```

Comment:
Since `resolve_location`, `derive_vault_id` and `derive_record_id`  no longer need to be methods, might be worth thinking about making them into functions for the Location API or something.   Calling `Self::` on these doesn't make sense when the input/output and logic don't really have anything to do with the `SecureClient`.   Not worth exposing them publicly from the client library but would clean up the code to at least just remove them from the impl block since they are effectively just wrapper functions.